wdl-format/tests/format/seaseq-case/source.formatted.wdl

```diff
@@ -0,0 +1,939 @@
+    ])
+
+    # FASTA faidx and chromsizes and effective genome size
+    call samtools.faidx as samtools_faidx {
+        # create FASTA index and chrom sizes files
+        input: reference = reference }
+    call util.effective_genome_size as egs {
+        # effective genome size for FASTA
+        input: reference = reference }
+
+    # Process FASTQs
+    if (defined(sample_fastq)) {
+
+        Array[String] string_fastq = [
+            1,
+        ]  #buffer to allow for fastq optionality
+        Array[File] s_fastq = select_first([
+            sample_fastq,
+            string_fastq,
+        ])
+
+        Array[File] sample_fastqfile = s_fastq
+    }
+    Array[File] original_fastqfiles = flatten(select_all([
+        sample_srafile,
+        sample_fastqfile,
+    ]))
+
+    ### ------------------------------------------------- ###
+    ### ---------------- S E C T I O N 1 ---------------- ###
+    ### ----------- B: remove Spike-IN reads ------------ ###
+    ### ------------------------------------------------- ###
+
+    # if multiple fastqfiles are provided
+    Boolean multi_fastq = if length(original_fastqfiles) > 1 then true else false
+    Boolean one_fastq = if length(original_fastqfiles) == 1 then true else false
+
+    if (defined(spikein_bowtie_index) || defined(spikein_reference)) {
+        scatter (eachfastq in original_fastqfiles) {
+            call fastqc.fastqc as spikein_indv_fastqc { input:
+                inputfile = eachfastq,
+                default_location = if (one_fastq) then sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/FastQC" else "SAMPLE/" + sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/FastQC",
+            }
+            call util.basicfastqstats as spikein_indv_bfs { input:
+                fastqfile = eachfastq,
+                default_location = if (one_fastq) then sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/SummaryStats" else "SAMPLE/" + sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/SummaryStats",
+            }
+            call bowtie.spikein_SE as spikein_indv_map { input:
+                fastqfile = eachfastq,
+                index_files = actual_spikein_bowtie_index,
+                metricsfile = spikein_indv_bfs.metrics_out,
+                default_location = if (one_fastq) then sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/SummaryStats" else "SAMPLE/" + sub(basename(eachfastq), ".fastq.gz|.fq.gz", "") + "/SpikeIn/SummaryStats",
+            }
+        }
+
+        Array[File] spikein_fastqfiles = spikein_indv_map.unaligned
+    }
+    Array[File] fastqfiles = select_first([
+        spikein_fastqfiles,
+        original_fastqfiles,
+    ])
+
+    ### ------------------------------------------------- ###
+    ### ---------------- S E C T I O N 2 ---------------- ###
+    ### ---- A: analysis if multiple FASTQs provided ---- ###
+    ### ------------------------------------------------- ###
+
```

Comment:
I would consider deleting this blank line.

Reply:
I think the other blanks you've pointed out I can come up with sane logic for deleting. This one not so much... If the blank on line 312 is permitted (and I think it should be), I also think this blank (on 317) must be permitted.

Reply:
I guess I think a blank line can precede a comment, but should we allow blanks to follow comments? I think we discussed this on Slack, but I just can't remember the conclusion. I don't necessarily want to rethink that.

Reply:
I don't think we discussed this specific case. I see a few different trivia conditions, that perhaps need to be handled individually (current code does not recognize anything I'm about to type)
- a blank line without comments before or after it
    - we've talked about this case over slack quite a bit, though I don't think we ever quite reached a conclusion on it
    - I believe I speak for both of us when I say there are cases where these should be removed and some cases they should stay
    - I just haven't figured out the heuristics for when they get removed yet
- a preceding comment [potentially a block of comments] without blanks before or after it
    - "internal blanks" (between comments) should always be respected (I'm fairly sure at least)
        - in case it wasn't clear, the "max one blank line" policy holds in all cases, including this one. Multiple blanks will **always** be consolidated into one blank
    - this is a simple case, because it is just copied without any "extra" blank padding
- a blank followed by a preceding comment (and no blank between the comment[s] and the non-trivia token)
    - I think this should be "left alone" i.e. the output is more or less the same
- the case we see above: a preceding comment [or block] that has a blank above and below
    - it seems you're saying the "below" blank should be deleted?
- final case: a preceding comment with only a blank below (none above)
    - probably this should be treated the same as the prior?

For context: the current code just always copies any blanks it finds. Which is not what I want it to be doing, but I haven't worked out the right approach for when blanks get removed.